test/integ/test_cursor.py

```diff
@@ -609,15 +609,119 @@ def test_geography(conn, db_parameters):
             cnx.cursor().execute(f"drop table {name_geo}")
 
 
+@pytest.mark.skipolddriver
 def test_callproc(conn_cnx):
-    """Callproc test.
+    """Callproc test."""
 
-    Notes:
-        It's a nop as of now.
-    """
+    def create_stored_procedure(cursor):
+        cursor.execute(
+            "drop procedure if exists test_stored_procedure(varchar, int, date)"
+        )
+        cursor.execute("drop procedure if exists test_stored_procedure(float, char)")
+        cursor.execute("drop procedure if exists test_stored_procedure()")
+
+        cursor.execute(
+            """
+            create or replace procedure test_stored_procedure(p1 varchar, p2 int, p3 date)
```

Comment:
You don't need to clean up the procedure if you make it temporary
```suggestion
            create or replace temporary procedure test_stored_procedure(p1 varchar, p2 int, p3 date)
```

Reply:
good suggestion, updated